### ISSUE TITLE
`(1|2|3)[]` type in diagnostics is printed as `1|2|3[]` instead of `(1|2|3)[]`

### ISSUE THREAD

**Issue by lochana-chathura** [ballerina-platform/ballerina-lang#35018]
**Description:**
```bal
public function main() {
    (1|2|3)[] ar = 3; // incompatible types: expected '1|2|3[]', found 'int'
}
```
As with type precedence, `'1|2|3|[]'` means `1|2|(3[])`.
Correct error msg should be `incompatible types: expected '(1|2|3)[]', found 'int'`

**Affected Versions:**
2201.0.0

**Comment by MaryamZi** [comment]
https://github.com/ballerina-platform/ballerina-lang/issues/32267

**Comment by shreyajha03** [comment]
@lochana-chathura Can I please take this issue? 
Thanks!

**Comment by lochana-chathura** [comment]
> @lochana-chathura Can I please take this issue? Thanks!
Sure! :)

**Comment by keizer619** [comment]
Hi @shreyajha03 ! 🚀

We value your contributions and look forward to collaborating with you.
To help you get started, here are some essential resources:

Understanding Ballerina Platform:
Learn what Ballerina is all about: [Ballerina Platform](https://github.com/ballerina-platform/ballerina-release/blob/master/CONTRIBUTING.md)
How to Contribute:
Read our contributing guidelines to understand how you can contribute effectively: [Contribution Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
Contributing to the Ballerina library:
If you're interested in contributing to our library, here are the guidelines: [Library Contribution Guide](https://github.com/ballerina-platform/ballerina-standard-library/blob/main/CONTRIBUTING.md)
Learn Ballerina:
If you're new to Ballerina or want to enhance your skills, our official website offers a wealth of learning resources: [Learn Ballerina](https://ballerina.io/learn/)
Get Help and Connect:
Join our Discord community to chat with fellow Ballerina enthusiasts and get assistance: [Ballerina Discord](https://discord.com/invite/ballerinalang)
Remember, no contribution is too small, and your feedback is invaluable. Feel free to ask questions, propose ideas, or report issues. Together, we can make Ballerina even better!
Happy coding! 🎉

**Comment by HeetVekariya** [comment]
@lochana-chathura If no is working on this issue, can i take this issue ?

**Comment by lochana-chathura** [comment]
@HeetVekariya I see that the issue has been assigned to @shreyajha03. If you could, please consider picking up another issue. Your help with different tasks would be greatly appreciated. Thank you.

**Comment by neelshah2409** [comment]
@lochana-chathura I would like to work on it as there are no updates

**Comment by m-tabish** [comment]
https://github.com/ballerina-platform/ballerina-library/issues/4764

**Comment by m-tabish** [comment]
Kindly assign this to me, under Hacktoberfest 2024. I will try to solve this

**Comment by Murdock9803** [comment]
Hello @lochana-chathura @keizer619 , I would like to take up the issue and provide help 🙌

**Comment by KasunUdaraWaththeHewa** [comment]
I need to fix this issue. please

**Comment by Ayahelmaoui** [comment]
can i also try to solve this?

**Comment by keizer619** [comment]
Hi @Murdock9803 @KasunUdaraWaththeHewa @m-tabish 
Since we have assigned other issues for you already, we will be assigning this to @Ayahelmaoui

**Comment by keizer619** [comment]
@Ayahelmaoui Are you working on this? If not we can open this for others

**Comment by Udara-Peiris** [comment]
Hi @keizer619 

This is my first time with Ballerina and I would like to work on this issue if anyone not work on this

**Comment by Sahasra0108** [comment]
@keizer619  can i also try to solve this?

**Comment by keizer619** [comment]
@Sahasra0108 Yes you can work on this issue if @Udara-Peiris is not working on this
